tabular/src/autogluon/tabular/models/xgboost/xgboost_model.py

```diff
@@ -152,3 +156,27 @@ def _get_early_stopping_rounds(self, num_rows_train, strategy='auto'):
 
     def _ag_params(self) -> set:
         return {'ag.early_stop'}
+
+    # FIXME: This is copy-pasted from CatBoostModel, make a generic memory check method / function and re-use.
+    def _assert_memory_safe(self, X, y):
+        num_rows_train = len(X)
+        num_cols_train = len(X.columns)
+        if self.problem_type == MULTICLASS:
+            if self.num_classes is not None:
+                num_classes = self.num_classes
+            else:
+                num_classes = 10  # Guess if not given, can do better by looking at y
+        elif self.problem_type == SOFTCLASS:  # TODO: delete this elif if it's unnecessary.
+            num_classes = y.shape[1]
+        else:
+            num_classes = 1
+        max_memory_usage_ratio = self.params_aux['max_memory_usage_ratio']
+        approx_mem_size_req = num_rows_train * num_cols_train * num_classes / 2  # TODO: Extremely crude approximation, can be vastly improved
+        if approx_mem_size_req > 1e9:  # > 1 GB
+            available_mem = psutil.virtual_memory().available
+            ratio = approx_mem_size_req / available_mem
+            if ratio > (1 * max_memory_usage_ratio):
+                logger.warning('\tWarning: Not enough memory to safely train XGBoost model, roughly requires: %s GB, but only %s GB is available...' % (round(approx_mem_size_req / 1e9, 3), round(available_mem / 1e9, 3)))
+                raise NotEnoughMemoryError
```

Comment:
Discussion point: would it be a good idea to sample and train on subset instead of no training at all?

Reply:
maybe, although often it's less about the number of rows and more about number of classes. I suspect we are better off training models that don't have to sample rows than sampling rows and training XGBoost in limited time scenarios.